File: src/main/java/ru/serdtsev/homemoney/account/ServiceAccount.java
package ru.serdtsev.homemoney.account;

import ru.serdtsev.homemoney.balancesheet.BalanceSheet;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.sql.Date;

@Entity
@DiscriminatorValue("service")
@Table(name = "svc_accounts")
public class ServiceAccount extends Account {
  protected ServiceAccount() {
  }

  public ServiceAccount(BalanceSheet balanceSheet, String name, Date created, Boolean isArc) {
    super(balanceSheet, AccountType.service, name, created, isArc);
  }
}

File: src/main/java/ru/serdtsev/homemoney/account/model/ServiceAccount.kt
package ru.serdtsev.homemoney.account.model

import ru.serdtsev.homemoney.balancesheet.BalanceSheet
import java.sql.Date
import java.util.*
import javax.persistence.DiscriminatorValue
import javax.persistence.Entity
import javax.persistence.Table

@Entity
@DiscriminatorValue("service")
@Table(name = "svc_accounts")
open class ServiceAccount(
        id: UUID,
        balanceSheet: BalanceSheet,
        name: String,
        createdDate: Date,
        isArc: Boolean?
) : Account(id, balanceSheet, AccountType.service, name, createdDate, isArc)